File: sdk/samples/WCFSamples/TechnologySamples/Scenario/PeerChannel/Chat/CS/CustomResolver/DefaultCustomResolver.cs
//  Copyright (c) Microsoft Corporation.  All Rights Reserved.

using System;
using System.Collections.Generic;
using System.Text;
using System.ServiceModel;
using System.ServiceModel.Channels;
using System.ServiceModel.PeerResolvers;
using System.Configuration;

namespace Microsoft.ServiceModel.Samples
{
    class DefaultCustomResolver
    {
        public static void Main()
        {
            // Create a new resolver service
            CustomPeerResolverService crs = new CustomPeerResolverService();
            crs.ControlShape = false;

            // Create a new service host
            ServiceHost customResolver = new ServiceHost(crs);

            // Open the resolver service 
            crs.Open();
            customResolver.Open();
            Console.WriteLine("Custom resolver service is started");
            Console.WriteLine("Press <ENTER> to terminate service");
            Console.ReadLine();
        }
    }
}

File: sdk/samples/WCFSamples/TechnologySamples/Scenario/PeerChannel/Chat/VB/CustomResolver/DefaultCustomResolver.vb
' Copyright (c) Microsoft Corporation.  All Rights Reserved.

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports System.ServiceModel
Imports System.ServiceModel.Channels
Imports System.ServiceModel.PeerResolvers
Imports System.Configuration

Namespace Microsoft.ServiceModel.Samples

    Class DefaultCustomResolver

        Public Shared Sub Main()

            ' Create a new resolver service
            Dim crs As New CustomPeerResolverService()
            crs.ControlShape = False

            ' Create a new service host
            Dim customResolver As New ServiceHost(crs)

            ' Open the custom resolver service and wait for user to press enter
            crs.Open()
            customResolver.Open()
            Console.WriteLine("Custom resolver service is started")
            Console.WriteLine("Press <ENTER> to terminate service")
            Console.ReadLine()

        End Sub

    End Class

End Namespace
